Gems/ONNX/Code/Source/Clients/ONNXSystemComponent.cpp

```diff
@@ -0,0 +1,300 @@
+        {
+            serialize->Class<ONNXSystemComponent, AZ::Component>()->Version(0);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ONNXSystemComponent>("ONNX", "[Description of functionality provided by this System Component]")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
+            }
+        }
+    }
+
+    void ONNXSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("ONNXService"));
+    }
+
+    void ONNXSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("ONNXService"));
+    }
+
+    void ONNXSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void ONNXSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    ONNXSystemComponent::ONNXSystemComponent()
+    {
+        if (ONNXInterface::Get() == nullptr)
+        {
+            ONNXInterface::Register(this);
+        }
+
+        m_timingStats.SetName("MNIST Timing Statistics");
+        m_timingStats.SetHistogramBinCount(200);
+
+        if (ENABLE_CUDA)
+        {
+            m_timingStatsCuda.SetName("MNIST CUDA Timing Statistics");
+            m_timingStatsCuda.SetHistogramBinCount(200);
+        }
+
+        ImGui::ImGuiUpdateListenerBus::Handler::BusConnect();
+    }
+
+    ONNXSystemComponent::~ONNXSystemComponent()
+    {
+        ImGui::ImGuiUpdateListenerBus::Handler::BusDisconnect();
+
+        if (ONNXInterface::Get() == this)
+        {
+            ONNXInterface::Unregister(this);
+        }
+    }
+
+    Ort::Env* ONNXSystemComponent::GetEnv()
+    {
+        return m_env.get();
+    }
+
+    Ort::AllocatorWithDefaultOptions* ONNXSystemComponent::GetAllocator()
+    {
+        return m_allocator.get();
+    }
+
+    void OnnxLoggingFunction(
+        void*, OrtLoggingLevel, const char* category, const char* logid, const char* code_location, const char* message)
+    {
+        AZ_Printf("\nONNX", "%s %s %s %s", category, logid, code_location, message);
+    }
+
+    // The global environment and memory allocator are initialised with the system component, and are accessed via the EBus from within the
+    // model. m_precomputedTimingData and m_precomputedTimingDataCuda are structs holding the test inference statistics run before the
+    // editor starts up, and used by the ImGui dashboard.
+    void ONNXSystemComponent::Init()
+    {
+        void* ptr;
+        m_env = AZStd::make_unique<Ort::Env>(ORT_LOGGING_LEVEL_VERBOSE, "test_log", OnnxLoggingFunction, ptr);
```

Comment:
ptr* is uninitialized and unnneded.
`nullptr` should passed as the parameter to initialize the [logger_param](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_env.html)
```suggestion
        m_env = AZStd::make_unique<Ort::Env>(ORT_LOGGING_LEVEL_VERBOSE, "test_log", OnnxLoggingFunction, nullptr);
```

Reply:
Fixed